examples/cpp/helloworld/README.md

```diff
@@ -256,5 +256,57 @@ main loop in `HandleRpcs` to query the queue.
 For a working example, refer to [greeter_async_server.cc](greeter_async_server.cc).
 
 
+#### Async bi-directional streaming server/client
 
+Bidirectional streaming RPCs allow both client and server to read and write 
+streaming data from/to each other, and this is supported by both synchronous and
+asynchronous API. The asynchronous API allows these actions to take place
+without blocking using the completion queue, just as in the asynchronous code
+examples above.
+One notable difference from the previous section on async RPCs and async
+streaming (bi-directional or otherwise) is the slightly complex nature of
+RPC flows and setup necessary:
 
+* Ensure the server calls `ServerContext::AsyncNotifyWhenDone(&tag)` during the
+  the stream setup so that you listen to notifications when the client is done 
+  with the stream.
+  
+* Ensure that the client calls `WritesDone` to indicate it is done with the
+  stream.
+
+* Ensure that client/server call `Finish` close the stream. This is an 
+  asymmetric API: The server `Finish`es the stream while the client listens
+  to the `Finish` notification generated by the server. This is an important 
+  distinction and is critical that both client/server handle this properly.
+
+A detailed example is in
+[greeter_async_bidi_server.cc](greeter_async_bidi_server.cc)
+and [greeter_async_bidi_client.cc](greeter_async_bidi_client.cc).
+
+
+#### Debugging gRPC
+
+To debug issues:
+
+```bash
+export GRPC_VERBOSITY=DEBUG
+```
+
+
+#### Completion queue usage options
+
+In the context of streaming servers/clients, several options are possible:
+
+ 1 Completion Queue  :: 1 Server :: 1 RPC (Simplest case)
+ 
+ 1 Completion Queue  :: 1 Server :: N RPCs (Sharing completion queue)
+ 
+ N Completion Queues :: 1 Server :: N RPCs (Independent completion queue
+  per RPC)
+  
+ A completion queue can also be shared among *different* RPCs as well, not
+  just for the same RPC API. In the context of streaming RPCs such as async
+  (bi-directional or uni-directional) streams, an RPC refers to a single stream
+  that is active until either side calls `Finish`.
+
```

Comment:
The first sentence of the above paragraph is confusing with regard to the second one because the first one is using RPC to mean "method" but the second one is using it to mean an actual invocation. The second one is helpful, of course.